src/it/resources/com/google/checkstyle/test/chapter4formatting/rule462horizontalwhitespace/InputNoWhitespaceBeforeAnnotations.java

```diff
@@ -0,0 +1,44 @@
+package com.google.checkstyle.test.chapter4formatting.rule462horizontalwhitespace;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Target;
+
+@Target(ElementType.TYPE_USE)
+@interface NonNull {
+
+}
+
+public class InputNoWhitespaceBeforeAnnotations {
+
+    public void foo(final char @NonNull [] param) {} // ok
+
+    @NonNull int @NonNull[] @NonNull[] fiel1; // OK until #8205
+    @NonNull int @NonNull [] @NonNull [] field2; // ok
+    //@NonNull int @NonNull ... field3; // non-compilable
+    //@NonNull int @NonNull... field4; // non-compilable
+
+
+    public void foo1(final char[] param) { // OK until #8205
+    }
+
+    public void foo2(final char [] param) { // ok
+    }
+
+    public void foo3(final char @NonNull[] param) { // OK until #8205
+    }
+
+    public void foo4(final char @NonNull [] param) { // ok
+    }
+
+    void test1(String... param) { // OK until #8205
+    }
+
+    void test2(String ... param) { // ok
```

Comment:
@romani I am not seeing where it is saying this expected/allowed.
https://checkstyle.org/styleguides/google-java-style-20180523/javaguide.html#s4.6.2-horizontal-whitespace
> a single ASCII space also appears in the following places only.
> Between a type annotation and [] or ....

There is no type annotation here and guide specifically says space is ONLY allowed where google defined it. There is no type annotation here, so there should be a violation.

https://github.com/checkstyle/checkstyle/issues/6707#issuecomment-602116682
> there is no item for space between type and [], so It is ok to demand no space before ....

It seems we are in agreement then? Nothing I see says this will be fixed later.

I am expecting changes to the logic of the check to look for annotation or not as specified in https://github.com/checkstyle/checkstyle/issues/6707#issuecomment-602065180

Reply:
> There is no type annotation here, so there should be a violation.

@rnveach, yes, you are right, I updated #8205 . We need to put here `// ok until #8205`.
@HuGanghui , please do the same for other similar cases.

We should remove token from Check config in google as it is error prone (it is better to not bother user with false-positives, no violation is much better for user experience). 
We will do new Check for this to address validation properly.

> I am expecting changes to the logic of the check to look for annotation or not as specified in

Lets move it #8205 , I think new Check should be created.

Reply:
done